src/resource_management.ts

```diff
@@ -36,22 +36,42 @@ export interface AsyncDisposable {
  */
 export function configureExplicitResourceManagement() {
   Symbol.asyncDispose &&
-    (MongoClient.prototype[Symbol.asyncDispose] = async function () {
-      await this.close();
+    Object.defineProperty(MongoClient.prototype, Symbol.asyncDispose, {
+      value: async function asyncDispose(this: { close(): Promise<void> }) {
+        await this.close();
+      },
+      enumerable: false,
+      configurable: true,
+      writable: true
     });
 
   Symbol.asyncDispose &&
-    (ClientSession.prototype[Symbol.asyncDispose] = async function () {
-      await this.endSession({ force: true });
+    Object.defineProperty(AbstractCursor.prototype, Symbol.asyncDispose, {
+      value: async function asyncDispose(this: { close(): Promise<void> }) {
+        await this.close();
+      },
+      enumerable: false,
+      configurable: true,
+      writable: true
     });
 
   Symbol.asyncDispose &&
-    (AbstractCursor.prototype[Symbol.asyncDispose] = async function () {
-      await this.close();
+    Object.defineProperty(ChangeStream.prototype, Symbol.asyncDispose, {
+      value: async function asyncDispose(this: { close(): Promise<void> }) {
+        await this.close();
+      },
+      enumerable: false,
+      configurable: true,
+      writable: true
     });
 
   Symbol.asyncDispose &&
-    (ChangeStream.prototype[Symbol.asyncDispose] = async function () {
-      await this.close();
+    Object.defineProperty(ClientSession.prototype, Symbol.asyncDispose, {
+      value: async function asyncDispose(this: { endSession(): Promise<void> }) {
+        await this.endSession();
```

Comment:
This was passing in `{force:true}` in the previous commit, I think you still want to do that correct? (Need to update the other copy of this as well)